### PR TITLE
docs: add readme

### PR DESCRIPTION
**Pull request by shreyanshJain852**
## Description

Add readme for the project description, design approach and installation

## Summary by CodeRabbit

- **Documentation**
	- Updated the naming convention in the README for better clarity and presentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,0 +1,77 @@
...
+> For requirements with any other level of access like for assigned resources, the implementation and handling for it, lies solely with the integrated application itself, which can be accomplished by using a generalized form of role and module access usage in the access control methods.
+
+## Setup Guidelines
+
+### Built With
+- [.NET 8 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
+- [Visual Studio 2022](https://visualstudio.microsoft.com/vs/)
+- [Docker](https://www.docker.com/products/docker-desktop/)
+
+### Setup
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md:37]: Headings are surrounded by blanks, no changes required here

File: README.md [hunk 2]
```diff
@@ -2,0 +1,77 @@
...
+> If the OsmoGuard module is running on the same machine / server as the integrated database, then the server name mentioned in database connection string has to match with the docker's host machine IP.
+>
+> The default IP assigned to host in docker is - `172.17.0.1`.
+>
+> The IP can be found by inspecting the bridge network using
+```docker inspect network bridge```.
+>
+> For Windows / Mac based machines, the host IP can also be represented by `host.docker.internal`. But for linux based machine, it has to the host IP only.
+
+# License
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md:69]: Headings are surrounded by blanks, no changes required here

File: README.md [hunk 3]
```diff
@@ -2,0 +1,77 @@
...
+This project is licensed under the [MIT](https://github.com/OsmosysSoftware/osmo-guard/blob/documentation/readme/LICENSE) license.
+## 👏 Big Thanks to Our Contributors
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md:71]: Headings are surrounded by blanks, no changes required here

File: README.md [hunk 4]
```diff
@@ -1 +1,77 @@
...
+It will be the responsibility of the integrated application to ensure that the access control database tables are not tampered with, and are not accessed directly from the application logic. The access related data is to be only requested from the access control module (OsmoGuard) and its results needs to be awaited for further processing any API request requiring role / permission related data.
+
+> [!NOTE]
+> Access control module database and its endpoint deals with generalized requirement for setting up roles, and their accessibility is limited to check for the create, read and update level access.
+>
+> For requirements with any other level of access like for assigned resources, the implementation and handling for it, lies solely with the integrated application itself, which can be accomplished by using a generalized form of role and module access usage in the access control methods.
+
+## Setup Guidelines
+
+### Built With
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md:32]: Headings are surrounded by blanks, no changes required here

File: README.md [hunk 5]
```diff
@@ -2,0 +1,77 @@
...
+- [.NET 8 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
+- [Visual Studio 2022](https://visualstudio.microsoft.com/vs/)
+- [Docker](https://www.docker.com/products/docker-desktop/)
+
+### Setup
+1. Clone the repository.
+    ```sh
+    git clone https://github.com/OsmosysSoftware/osmo-guard.git
+    ```
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md:41]: The code block is indented under the 1st point with proper spaces

File: README.md [hunk 6]
```diff
@@ -2,0 +1,77 @@
...
+    cd osmo-guard/
+    ```
+3. Update the `src\OsmoGuard.API\appsettings.json` file, with the database connection string used in the project.
+
+    Note: The OmsoGuard module is setup with MySQL database integration. To change the database server used, for the application, changes will be required in the `src\OsmoGuard.API\program.cs`.
+
+4. Run the application with docker compose.
+    ```sh
+    docker compose up -d
+    ```
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md:53]: The code block is indented under the 4th point with proper spaces

File: README.md [hunk 7]
```diff
@@ -2,0 +1,77 @@
...
+5. The application server is available at
+    ```
+    http://localhost:5000/
+    ```
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md:57]: The code block is indented under the above point with proper spaces

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md:69]: The heading License, already has a blank line above it for formatting

File: README.md [hunk 8]
```diff
@@ -2,0 +1,77 @@
...
+
+OsmoGuard is based on RBAC (role based access control) method, which defines roles and their corresponding privileges within an application domain. Every role has a collection of permissions and restrictions. A user can access objects and execute operations only if their role has the relevant permissions.
+
+## Design Approach
+
+OsmoGuard is a dockerized project-dependent microservice to be deployed along with the project server.
+
+It will be responsible for maintaining and tracking the user roles and their accessibility across the application domain and resources.
+
+It will expose endpoints for setting up the the modules, roles and their privileges, which will be retrieved and utilized from the authorization middleware of the application.
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply README.md]: updated

**Review by sameer-s-b — APPROVED** [review]

**Pull request merged**